WEB: Added theme dependent favicon

DeviousLab opened the pull request:
- [x] closes #42314
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

Added a light mode version of the favicon based upon the .svgs in the static folder as well as added a small js scipt to the template page to detect the theme change.

![mgCmqcreiy](https://user-images.githubusercontent.com/53790060/123965934-fe1c8a00-d9c5-11eb-8838-ce72ee988a3c.gif)

DeviousLab committed:
Added white favicon for dark mode

DeviousLab committed:
Added script to change ico upon theme

attack68 commented on web/pandas/_templates/layout.html:
```diff
@@ -99,5 +100,34 @@
+            }
+            matcher.addListener(onUpdate);
+            onUpdate();
+            }
+
+            setupIcons();
```
```suggestion
        }

        setupIcons();
```

attack68 commented on web/pandas/_templates/layout.html:
```diff
@@ -99,5 +100,34 @@
                 crossorigin="anonymous"></script>
+        <script type="text/javascript">
+        function setupIcons() {
+            const lightSchemeIcon = document.querySelector('link#light-scheme-icon');
```
```suggestion
            // change the favicon based on browser detected color scheme on refresh
            const lightSchemeIcon = document.querySelector('link#light-scheme-icon');
```

DeviousLab committed:
Added suggested indenting

Co-authored-by: attack68 <24256554+attack68@users.noreply.github.com>

DeviousLab committed:
Described with comment what the script does

Co-authored-by: attack68 <24256554+attack68@users.noreply.github.com>

attack68 reviewed (approved):
lgtm.

jreback commented:
cc @datapythonista if any comment

datapythonista reviewed (changes requested):
Thanks for the contribution @DeviousLab. Looks fine, but I think we can get a way simpler implementation. Your code is only executed at load time, so why not leave the html as is (just adding an id), and the javascript simply have an if to check if the theme is dark, and if it change the href value?

DeviousLab committed:
Condensed js to update href

Simplified code as requested, now updates the favicon without requiring refresh through use of a event listener. And now just updates the href of the current favicon.

DeviousLab commented:
@datapythonista I probably have done it this way in the first place, but I've made the changes you've requested, now updates the favicon without requiring refresh through use of a event listener. And now just updates the href of the current favicon. Definitely a lot simpler. 
![363GRl3Xq6](https://user-images.githubusercontent.com/53790060/124248105-91270280-db33-11eb-804c-2aa41866ed93.gif)

datapythonista reviewed (approved):
I don't think a theme update is something we need to worry about, but in any case, this is much clearer, looks good to me.

Thanks for working on it @DeviousLab

DeviousLab commented:
Are there any more changes that need to be made? Or can it be safely merged without conflicts?

datapythonista commented:
Sorry for the delay @DeviousLab. We usually require two core devs to review a PR before merging, let's wait a bit more and see if someone else can have a look. Thanks for your patience.

DeviousLab commented:
> Sorry for the delay @DeviousLab. We usually require two core devs to review a PR before merging, let's wait a bit more and see if someone else can have a look. Thanks for your patience.

No worries! just thought I might have been missing something

DeviousLab commented:
@jreback Does this look good to you?

jreback commented:
thanks @DeviousLab
